src/query/expression/src/types/binary.rs

```diff
@@ -456,6 +457,24 @@ impl BinaryColumnBuilder {
         self.data.get_unchecked(start..end)
     }
 
+    pub fn push_repeat(&mut self, item: &[u8], n: usize) {
+        self.data.reserve(item.len() * n);
```

Comment:
`builder.reserve(self.data.len() + item.len() * n)`.

Reply:
## pub fn [reserve](https://doc.rust-lang.org/beta/alloc/vec/struct.Vec.html#method.reserve)(&mut self, additional: [usize](https://doc.rust-lang.org/beta/core/primitive.usize.html))

Reserves capacity for at least additional more elements to be inserted in the given Vec<T>. The collection may reserve more space to speculatively avoid frequent reallocations.

After calling reserve, capacity will be greater than or equal to `self.len() + additional`.

Does nothing if capacity is already sufficient.

## Examples
```rust
// len: 1, cap:1, cap-len:0 no space for more element
let mut vec = vec![1]; 

// reserves additional 10,
// then capacity >=11 ( 1(len)+10(additional) )
// so we can insert 10 additional more elements without allocation
vec.reserve(10);  
assert!(vec.capacity() >= 11);
```
## Src
```rust
impl<T, A: Allocator> RawVec<T, A> {
    /// Returns if the buffer needs to grow to fulfill the needed extra capacity.
    /// Mainly used to make inlining reserve-calls possible without inlining `grow`.
    fn needs_to_grow(&self, len: usize, additional: usize) -> bool {
        additional > self.capacity().wrapping_sub(len)
    }
```